### ISSUE TITLE
cupsd.conf | <Limit operation>/<Limit method> does not limit the access to operations or methods

### ISSUE THREAD

**Issue by felixwttr** [OpenPrinting/cups#1018]
I want to limit the access to operations like Pause-Printer, Disable-Printer, ... to a specific IP address.

**CUPS** version: 2.4.2
**OS**: Debian 12

My **cupsd.conf** file looks like this:

```
...
<Location />
  # Allow remote access...
  Order allow,deny
  Allow all
</Location>
<Location /admin>
  Allow from xxx.xxx.xxx.xxx
  Encryption Required
  Order allow,deny
</Location>
<Location /admin/conf>
  Allow from xxx.xxx.xxx.xxx
  Encryption Required
</Location>
<Location /printers>
  Order allow,deny
  Allow All
  <Limit POST>
    Allow from xxx.xxx.xxx.xxx
  </Limit>
</Location>

<Policy default>
  JobPrivateAccess all
  JobPrivateValues none
  SubscriptionPrivateAccess all
  SubscriptionPrivateValues none
   <Limit Create-Job Print-Job Print-URI Validate-Job>
     Order deny,allow
   </Limit>
   <Limit Send-Document Send-URI Hold-Job Release-Job Restart-Job Purge-Jobs Set-Job-Attributes Create-Job-Subscription Renew-Subscription Cancel-Subscription Get-Notifications Reprocess-Job Cancel-Current-Job Suspend-Current-Job Resume-Job Cancel-My-Jobs Close-Job CUPS-Move-Job CUPS-Get-Document>
     Allow from xxx.xxx.xxx.xxx
     Order allow,deny
   </Limit>
   <Limit CUPS-Add-Modify-Printer CUPS-Delete-Printer CUPS-Add-Modify-Class CUPS-Delete-Class CUPS-Set-Default CUPS-Get-Devices>
     Allow from xxx.xxx.xxx.xxx
     Order allow,deny
   </Limit>
   <Limit Pause-Printer Resume-Printer Enable-Printer Disable-Printer Pause-Printer-After-Current-Job Hold-New-Jobs Release-Held-New-Jobs Deactivate-Printer Activate-Printer Restart-Printer Shutdown-Printer Startup-Printer Promote-Job Schedule-Job-After Cancel-Jobs CUPS-Accept-Jobs CUPS-Reject-Jobs>
     Allow from xxx.xxx.xxx.xxx
     Order allow,deny
     Satisfy all
   </Limit>
   <Limit Cancel-Job CUPS-Authenticate-Job>
     Allow from xxx.xxx.xxx.xxx
     Order allow,deny
   </Limit>
  <Limit Pause-Printer>
    Allow from xxx.xxx.xxx.xxx
    Order allow,deny
  </Limit>
  <Limit All>
    Allow from xxx.xxx.xxx.xxx
    Order allow,deny
  </Limit>
</Policy>
DefaultPolicy default
``` 

For the Locations it's working and "/" is open for any IP while "/admin" is only accessable for the specified IP.

The Limit however is not working. Not in the Location where I tried to limit the POST requestes and not in the Policies where I wanted to limit certain IPP methods.

Can anybody assist on where my issue is or is this behavior a Bug in CUPS?

Thanks in advance :)
Regards
Felix

**Comment by zdohnal** [comment]
@felixwttr Please define what do you mean by "not working" - does it mean all machines can do the limited method/operation? Or none is capable of doing them?

I've checked <Limit GET> in <Location /printers> and <Limit CUPS-Get-Printers> under default policy, and it works with version 2.4.10. Can you check whether it does not work with version 2.4.10?

**Comment by felixwttr** [comment]
Thanks for your reply @zdohnal :)

By "not working" I mean, that if I define my config like mentioned above all machines can execute the limited method/operation.

Unfortunately don't have any CUPS with version 2.4.10 and upgrading is not possible currently, because I am in a corporate environment. 

But I just found a configuration that works.

```
<Location /printers>
  Order deny,allow
  <Limit POST>
    Allow from xxx.xxx.xxx.xxx
    Order allow,deny
  </Limit>
</Location>
``` 

With that configuration no HTTP POST is not possible from any machine except xxx.xxx.xxx.xxx inside the /printer location.

Maybe there was a bug in 2.4.2 and it's fixed in 2.4.10.
